src/generators/operationGenerator.ts

```diff
@@ -381,46 +381,29 @@ function getOperationParameterSignatures(
 function getBaseMethodParameterDeclarations(
   overloadParameterDeclarations: ParameterWithDescription[][]
 ): ParameterWithDescription[] {
-  const baseMethodParameters: ParameterWithDescription[] = [];
-  // Need to know which overload has the most parameters to set our upper bound.
-  const maxOverloadSize = Math.max(
-    ...overloadParameterDeclarations.map(params => params.length)
-  );
-  for (let i = 0; i < maxOverloadSize; i++) {
-    // attempt to combine types
-    const declarations: ParameterWithDescription[] = [];
-    for (const overloadParameterDeclaration of overloadParameterDeclarations) {
-      if (overloadParameterDeclaration[i]) {
-        declarations.push(overloadParameterDeclaration[i]);
-      }
-    }
-
-    const declaration = declarations.reduce(
-      (prevDeclaration, curDeclaration) => {
-        prevDeclaration = { ...prevDeclaration };
-        if (curDeclaration.name !== prevDeclaration.name) {
-          // Currently we only generate overloads if an operation supports multiple media types.
-          // Since contentType is always required and the 1st parameter, parameter names/ordering
-          // shouldn't change.
-          // In order to support more variance in parameter naming/ordering, we'll need to be able
-          // to construct the OperationArguments separately for each overload.
-          throw new Error(
-            `Operation overloads with different parameter names/ordering not supported.`
-          );
-        }
-        if (curDeclaration.type !== prevDeclaration.type) {
-          prevDeclaration.type += ` | ${curDeclaration.type}`;
-        }
-        // parameters should be optional if any declaration is optional
-        if (!curDeclaration.hasQuestionToken) {
-          prevDeclaration.hasQuestionToken = curDeclaration.hasQuestionToken;
-        }
-        return prevDeclaration;
-      }
-    );
-    baseMethodParameters.push(declaration);
+  if (!overloadParameterDeclarations.length) {
+    return [];
   }
-  return baseMethodParameters;
+  if (overloadParameterDeclarations.length === 1) {
+    return [...overloadParameterDeclarations[0]];
+  }
+
+  const baseMethodArg: ParameterWithDescription = {
+    name: "args",
+    isRestParameter: true,
+    description: "Includes all the parameters for this operation.",
+    type: overloadParameterDeclarations
+      .map(overloadParams => {
+        return `[ ${overloadParams
+          .map(param => {
+            return param.hasQuestionToken ? `${param.type}?` : param.type;
```

Comment:
not important - you could remove the braces and return inside this map

Reply:
I was able to get it to fit on a single line by changing `param` to `p` as well, so made the change.